source/common/buffer/buffer_impl.h

```diff
@@ -749,6 +753,30 @@ class OwnedImpl : public LibEventInstance {
   void commit(uint64_t length, absl::Span<RawSlice> slices,
               ReservationSlicesOwnerPtr slices_owner) override;
 
+  uint8_t* inlineReserve(uint64_t size) override {
+    // If no any slice in the buffer then create one new slice based on the expected continuous
+    // memory size.
+    if (slices_.empty()) {
+      slices_.emplace_back(Slice(size, account_));
+      length_ += size;
+      slices_.back().reservable_ += size;
```

Comment:
The "length_ += size;" and "slices_.back().reservable_ += size;" in here are a bit odd.  inlineReserve is not only reserving, but also committing bytes to the buffer before the caller of inlineReserve has written the bytes to the appropriate buffer.

I take it that this is an optimization.  I think that the method name should be changed to reflect this reserve+commit semantic that your implementation depends on.

Reply:
get it.